glotaran/deprecation/deprecation_utils.py

```diff
@@ -0,0 +1,504 @@
+"""Helper functions to give deprecation warnings."""
+
+from __future__ import annotations
+
+import os
+import sys
+from functools import wraps
+from importlib import import_module
+from importlib.metadata import distribution
+from types import ModuleType
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Callable
+from typing import TypeVar
+from typing import cast
+from warnings import warn
+
+DecoratedCallable = TypeVar(
+    "DecoratedCallable", bound=Callable[..., Any]
+)  # decorated function or class
+
+if TYPE_CHECKING:
+    from typing import Sequence
+
+
+class OverDueDeprecation(Exception):
+    """Error thrown when a deprecation should have been removed.
+
+    See Also
+    --------
+    deprecate
+    warn_deprecated
+    deprecate_module_attribute
+    deprecate_submodule
+    """
+
+    pass
+
+
+def glotaran_version() -> str:
+    """Pyglotaran version of the distribution.
+
+    This is basically the same as ``glotaran.__version__`` but independent from glotaran.
+    This way all of the deprecation functionality can be used even in
+    ``glotaran.__init__.py`` without moving the import below the definition of
+    ``__version__`` or causeing a circular import issue.
+
+    Returns
+    -------
+    str
+        Versionstring of glotaran.
+    """
+    return distribution("pyglotaran").version
+
+
+def parse_version(version_str: str) -> tuple[int, int, int]:
```

Comment:
This doesn't seem to take the -qualifier into account (e.g. -dev -develop -development). 
What if we release 0.5.0 and now change the version sources to 0.6.0-develop … will the stuff that was marked for deprecation now immediately break? or is -dev still allowed?
0.6.0-develop should be treated as 0.5.x+1 (where x is the last minor released version) or if deprecation can only happen in minor version bumps anyways, 0.6.0-develop is just 0.5.0 as far as the deprecation API is concerned

Reply:
Addressed in [this commit ](https://github.com/glotaran/pyglotaran/pull/631/commits/1c5ded3912463db868521f8d39aaf5d1daf42945)